### DISCUSSION TITLE
Changing all stock items to not delete on depletion

### DISCUSSION THREAD

**Discussion by StormTrooper** [inventree/InvenTree#4437] [Q&A] [answered]
I've created a few hundred stock items and now realised I should have disabled the delete on deplete option. 

I found a mysql table with this setting and wondering if it's safe to just change this parameter or will there be other isses that I"m not aware of if I do?
Is it as simple as updating the records to 0?

stock item with option disabled:
```
mysql> select id, delete_on_deplete from stock_stockitem where id='14';
+----+-------------------+
| id | delete_on_deplete |
+----+-------------------+
| 14 |                 0 |
+----+-------------------+
1 row in set (0.00 sec)
```

stock item with option enabled:
```
mysql> select id, delete_on_deplete from stock_stockitem where id='15';
+----+-------------------+
| id | delete_on_deplete |
+----+-------------------+
| 15 |                 1 |
+----+-------------------+
1 row in set (0.00 sec)
```

**Comment by matmair** [comment]
@StormTrooper in general I do not recommend directly writing to the database. In this case, it should be safe. Maybe:
* turn off the server before doing it
* run it in one SQL statement

to ensure the transaction is atomic.

**Comment by SchrodingersGat** [comment, marked as answer]
@StormTrooper if you have shell access to the server you can do the same thing through the django ORM CLI:

```
cd InvenTree
python ./manage.py shell

>>> from stock.models import StockItem()
>>> StockItem.objects.all().update(delete_on_deplete=False)
```

**Comment by StormTrooper** [comment]
Thanks guys. I didn't know about the shell - this has set the  option. 

I had to use ```from stock.models import StockItem``` though. Maybe just a typo?

**Reply by SchrodingersGat** [reply]
Yep, typo! Sorry about that
